frontend/javascripts/oxalis/model/sagas/task_saga.js

```diff
@@ -124,3 +131,58 @@ export default function* watchTasksAsync(): Saga<void> {
     yield* put(setActiveUserAction(fullUser));
   }
 }
+
+export function* warnAboutMagRestriction(): Saga<void> {
+  function* warnMaybe(): Saga<void> {
+    const { allowUpdate } = yield* select(state => state.tracing.restrictions);
+    if (!allowUpdate) {
+      // If updates are not allowed in general, we return here, since we don't
+      // want to show any warnings when the user cannot edit the annotation in the first
+      // place (e.g., when viewing the annotation of another user).
+      return;
+    }
+
+    const isViolated = yield* select(isMagRestrictionViolated);
+    const toastConfig = { sticky: true, key: "mag-restriction-warning" };
+
+    if (isViolated) {
+      const [min, max] = yield* select(storeState => getValidTaskZoomRange(storeState, true));
+      const clampZoom = () => {
+        const currentZoomStep = Store.getState().flycam.zoomStep;
+        const newZoomValue = clamp(min, currentZoomStep, max);
+        Store.dispatch(setZoomStepAction(newZoomValue));
+      };
+      const message = (
+        <React.Fragment>
+          Annotating data is restricted to a certain zoom range. Please adapt the zoom value so that
+          is between {min.toFixed(2)} and {max.toFixed(2)}. Alternatively, click{" "}
+          <Button type="link" onClick={clampZoom}>
+            here
+          </Button>{" "}
+          to adjust the zoom accordingly.
+        </React.Fragment>
+      );
+
+      Toast.error(message, toastConfig);
+    } else {
+      Toast.close(toastConfig.key);
+    }
+  }
+
+  yield* take("WK_READY");
+  // Wait before showing the initial warning. Due to initialization lag it may only be visible very briefly, otherwise.
+  yield _delay(5000);
+  yield* warnMaybe();
+
+  while (true) {
+    yield* take([
+      "ZOOM_IN",
+      "ZOOM_OUT",
+      "ZOOM_BY_DELTA",
+      "SET_ZOOM_STEP",
+      "SET_STORED_LAYOUTS",
+      "SET_TOOL",
```

Comment:
Good catch regarding the `SET_TOOL` action :+1: This is a leftover from a version where the saga switched the tool (instead of the volume tools view). I'll remove it.

> Also, if I remember correctly, the zoom step could also be influenced by the SET_INPUT_CATCHER_RECT and SET_INPUT_CATCHER_RECTS actions.

I just verified that `SET_STORED_LAYOUTS` is called when resizing the viewports and when toggling the fullscreen mode. So, the other actions should not be necessary to listen to. Let me know if other cases come to your mind where it could be important, too.

Probably, the store definition for the input_catcher_rects and the layouts could use some clean up, as their semantics overlap here. However, I'm hesitant to touch that code :see_no_evil:

Reply:
> I just verified that SET_STORED_LAYOUTS is called when resizing the viewports and when toggling the fullscreen mode. So, the other actions should not be necessary to listen to. Let me know if other cases come to your mind where it could be important, too.

Ah okay, cool, thanks for checking :+1: